apps/aecore/lib/aecore/naming/name_util.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Aecore.Naming.NameUtil do
+  @moduledoc """
+  Aecore naming utilities.
+  """
+
+  alias Aeutil.Hash
+
+  @split_name_symbol "."
+
+  @name_registrars [@split_name_symbol <> "aet", @split_name_symbol <> "test"]
+
+  @spec normalized_namehash!(String.t()) :: binary()
+  def normalized_namehash!(name) do
+    case normalize_and_validate_name(name) do
+      {:ok, normalized_name} -> namehash(normalized_name)
+      {:error, error} -> throw(error)
```

Comment:
This would lead to very deep nesting for correct error handling, as I tried implementing this. Maybe you can propose a solution for this, which is more elegant.

Reply:
Okay, here is my proposal. I see that you want to check the validity of the name before doing any of the NamingTxs, but sometimes you check it at creation time, and sometimes not. I propose to check it only in creation time with a `case` statement. To escape the repeated `case` statement you could make a generic naming_tx that will get the payload, check the naming and create a `data_tx` followed by a `spend_tx`. I know that you have 2 cases where in the payload the name has a different key. Can't you make it the same throughout the naming transaction payloads?

Reply:
can you add a commit for this, otherwise I'd leave it like this and maybe adjust in the future?

Reply:
I'll add a commit with my suggestion.